app/javascript/components/track/approaches-elements/community-videos/CommunityVideo.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import { Avatar, Icon } from '@/components/common'
+import { CommunityVideo } from './types'
+import { CommunityVideoModal } from './CommunityVideoModal'
+
+export function CommunityVideo({
+  video,
+}: {
+  video: CommunityVideo
+}): JSX.Element {
+  const [isOpen, setIsOpen] = useState(false)
+
+  return (
+    <div className="flex">
+      <button
+        onClick={() => setIsOpen(true)}
+        className="flex items-center justify-between bg-white shadow-sm rounded-8 px-20 py-16 mb-16 grow"
+      >
+        <div className="flex items-center">
+          <img
+            style={{ objectFit: 'cover', height: '80px', width: '143px' }}
+            className="mr-20 rounded-8"
+            src={video.links.thumbnail}
+            alt="thumbnail"
+          />
+          <div className="flex flex-col">
+            <h5 className="text-h5 mb-8">{video.title}</h5>
+            <div className="flex flex-row items-center">
+              {video.author && (
+                <Avatar
+                  src={video.author.avatarUrl}
+                  className="mr-8 h-[24px] w-[24px]"
+                />
+              )}
+              <span className="font-semibold text-textColor6 leading-150 text-14">
+                {video.author && video.author.name}
+              </span>
+            </div>
+          </div>
+        </div>
+
+        <Icon
+          className="filter-textColor6 h-[24px] w-[24px]"
+          icon={'expand'}
+          alt={'see video'}
+        />
+      </button>
+      <CommunityVideoModal
```

Comment:
I don't really like this. This is a simple way of doing it, and probably the easiest to maintain. But it creates many ModalPortals, which might be empty when unopened, but still I feel it'd be nicer with one portal which changes its content. idk.

Reply:
Maybe create an issue for it?